mutable_tree.go

```diff
@@ -501,3 +501,50 @@ func (tree *MutableTree) addOrphans(orphans []*Node) {
 		tree.orphans[string(node.hash)] = node.version
 	}
 }
+
+// IMPORTANT This function assumes there has been no transactions since the last SaveVersion. If unsure about
+// this do a SaveVersion immediately before calling this function.
+//
+// NewSliceAt creates a copy of the current tree consisting only of the given version.
+// Uses input db as a new backing database.
+func (tree *MutableTree) NewSliceAt(version int64, newDb dbm.DB) ([]byte, int64, *MutableTree, error) {
```

Comment:
Lets call this `SaveVersionToDB`, and it should return whatever `tempTree.SaveVersion()` returns